eclipse-che/che-server
# Fetch oauth tokens from kubernetes secrets

### What does this PR do?
1. Move `AuthorisationRequestManager` interface to`wsmaster/che-core-api-factory` module in order to avoid circular dependency.
2. Rework the get oauth token method to return an oauth token from kubernetes secrets if token is not found in the system memory of the `che-server` pod. If `provider name` value is not configured in the kubernetes secret oauth token, get the token by mapping it by provider url from the secret and oauth provider name from the aouth config secret.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/22776

### How to test this PR?

1. Deploy che from the PR image: `quay.io/eclipse/che-server:pr-652`
2. Configure [GitHub oauth](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-github/)
3. Start a workspace from the GitHub repository with a devfile.
4. Restart the `che-server` pod.
5. Go to the dashboard -> User Preferences -> Git Services and revoke the GitHub authorisation.

See: The authorisation revoke action is executed successfully.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
